lib/ios-deploy.js

```diff
@@ -204,22 +205,62 @@ class IOSDeploy {
       }
       const executableName = apps[bundleId].CFBundleExecutable;
       log.debug(`The executable name for the bundle id '${bundleId}' was '${executableName}'`);
-      instrumentService = await services.startInstrumentService(this.udid);
-      const processes = await instrumentService.callChannel(
-        INSTRUMENT_CHANNEL.DEVICE_INFO,
-        'runningProcesses',
-      );
-      const process = processes.selector.find((process) => process.name === executableName);
-      if (!process) {
-        log.info(`The process of the bundle id '${bundleId}' was not running`);
-        return false;
+
+      // 'devicectl' has overhead than the instrument service via appium-ios-device,
+      // so hre uses the 'devicectl' only for iOS 17+.
+      if (util.compareVersions(platformVersion, '>', '17.0')) {
+        log.debug(`Calling devicectl to kill the process`);
+
+        // FIXME: replace `devicectl` command with a wrapper later
+        // or remove after implementing appium-ios-device for iOS 17+.
+
+        const xcrunBinnaryPath = await requireXcrun();
+        let args = [];
+        args.push(
+          'devicectl',
+          'device',
+          'info',
+          'processes',
+          `--device`, this.udid
+        );
+        const {stdout} = await exec(xcrunBinnaryPath, args);
```

Comment:
I believe the tool has an option to spit JSON formatted output instead

Reply:
yea, it has an option `--json-output` to a file (file path was required), so currently parse the stdoutput vs output as JSON, and read it. So far, I chosed the parsing the output way for now since this output is simple.